azure/services/managedclusters/spec.go

```diff
@@ -437,6 +435,16 @@ func (s *ManagedClusterSpec) Parameters(ctx context.Context, existing interface{
 		// AgentPool changes are managed through AMMP.
 		managedCluster.AgentPoolProfiles = existingMC.AgentPoolProfiles
 
+		// if the AuthorizedIPRanges is nil in the user-updated spec, but not nil in the existing spec, then
+		// we need to set the AuthorizedIPRanges to empty array (&[]string{}) once so that the Azure API will
+		// update the existing authorized IP ranges to nil.
+		if !apiServerAccessProfileOrAuthIPRangesIsNil(existingMC) && apiServerAccessProfileOrAuthIPRangesIsNil(managedCluster) {
+			log.V(4).Info("managed cluster spec has nil AuthorizedIPRanges, updating existing authorized IP ranges to empty")
+			managedCluster.APIServerAccessProfile = &containerservice.ManagedClusterAPIServerAccessProfile{
+				AuthorizedIPRanges: &[]string{},
+			}
+		}
```

Comment:
Heads up about test coverage, maybe add something in spec_test.go?

Reply:
I have added more tests to spec_test.go covering scenarios testing this scenario. Wdyt?

Reply:
Looks good!